interface-impl-amf/src/main/java/org/mule/amf/impl/AMFParser.java

```diff
@@ -6,83 +6,88 @@
+import java.util.concurrent.ScheduledExecutorService;
 
-import amf.ProfileName;
-import amf.client.AMF;
-import amf.client.environment.DefaultEnvironment;
-import amf.client.environment.Environment;
-import amf.client.model.document.BaseUnit;
-import amf.client.model.document.Document;
-import amf.client.model.domain.WebApi;
-import amf.client.parse.Parser;
-import amf.client.resolve.Raml10Resolver;
-import amf.client.validate.ValidationReport;
+import static java.util.stream.Collectors.toList;
+import static org.mule.amf.impl.DocumentParser.getParserForApi;
+import static org.mule.amf.impl.DocumentParser.getParsingReport;
+import static org.mule.amf.impl.URIUtils.getPathAsUri;
 
 public class AMFParser implements ApiParser {
 
+  private static final Logger logger = LoggerFactory.getLogger(DocumentParser.class);
+
   private ApiReference apiRef;
-  private Parser parser;
+  private AMFParserWrapper parser;
   private WebApi webApi;
   private List<String> references;
 
   private Document consoleModel;
   private ExecutionEnvironment executionEnvironment;
 
-  public AMFParser(ApiReference apiRef, boolean validate, ScheduledExecutorService scheduler) throws ExecutionException, InterruptedException {
+  public AMFParser(ApiReference apiRef, boolean validate, ScheduledExecutorService scheduler) {
     initialize(apiRef, validate, new ExecutionEnvironment(scheduler));
   }
 
-  public AMFParser(ApiReference apiRef, boolean validate) throws ExecutionException, InterruptedException {
+  public AMFParser(ApiReference apiRef, boolean validate) {
     initialize(apiRef, validate, new ExecutionEnvironment());
   }
 
-  private void initialize(ApiReference apiRef, boolean validate, ExecutionEnvironment executionEnvironment) throws ExecutionException, InterruptedException {
-    AMF.init(executionEnvironment).get();
+  private void initialize(ApiReference apiRef, boolean validate, ExecutionEnvironment executionEnvironment) {
     this.executionEnvironment = executionEnvironment;
     this.apiRef = apiRef;
     this.parser = initParser(apiRef);
 
     Document document = buildDocument(validate);
     this.references = getReferences(document.references());
-    this.webApi = DocumentParser.getWebApi(document);
+    this.webApi = (WebApi) document.encodes();
   }
 
-  private Parser initParser(ApiReference apiRef) {
+  private AMFParserWrapper initParser(ApiReference apiRef) {
     final Environment environment = buildEnvironment(apiRef);
-    return getParserForApi(apiRef, environment, executionEnvironment);
+    try {
+      if (executionEnvironment != null) {
+        AMF.init(executionEnvironment).get();
+      } else {
+        AMF.init().get();
+      }
+//      AMFValidatorPlugin.withEnabledValidation(true);
+      amf.core.AMF.registerPlugin(new XmlValidationPlugin());
+    } catch (final Exception e) {
+      logger.error("Error initializing AMF", e);
+      throw new RuntimeException(e);
+    }
+    return getParserForApi(apiRef, environment);
```

Comment:
why not return AMFParserWrapperFactory.getParser(apiRef, environment);

Reply:
Updated